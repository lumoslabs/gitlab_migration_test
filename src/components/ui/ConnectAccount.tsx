import React from 'react'
import { StyleSheet, css } from 'aphrodite'
import { Container, Row, Col } from 'react-bootstrap'
import base from '@styles/colors/base'
import commonStyles from '@styles/commonStyles'
import Button from '@components/ui/Button'

export interface IConnectAccountProps {
  handleCancel(e: React.MouseEvent<any>): any;
  handleConnect(e: React.MouseEvent<any>): any;
}

const ConnectAccount = (props: IConnectAccountProps): JSX.Element => {
  const { handleCancel, handleConnect } = props

  return (
    <Container className={css(commonStyles.flexColumnAlignCenter)}>
      <Row className={css(commonStyles.flexRowAllCenter)}>
        <p className={css(styles.title)}>
          By clicking Connect, Google Assistant will collect your Lumosity audio commands,
          and their Terms and Privacy Policy will also apply along with Lumosity’s.
          <br />
          <br />
          {"To remove Google Assistant’s access, please visit Lumosity's Help Center."}
        </p>
      </Row>
      <Row className={css(commonStyles.flexRowAllCenter)}>
        <Col className={css(commonStyles.flexColumnAllCenter, styles.buttonDiv)}>
          <Button onClick={handleCancel} text='Cancel' />
        </Col>
        <Col className={css(commonStyles.flexColumnAllCenter, styles.buttonDiv)}>
          <Button onClick={handleConnect} text='Connect' />
        </Col>
      </Row>
    </Container>
  )
}

export default ConnectAccount

const styles = StyleSheet.create({
  title: {
    color: base.lumosBlack,
    fontFamily: 'MuseoSans500',
    fontSize: '26px',
    fontWeight: 700,
    display: 'flex',
    paddingRight: '20vh',
    paddingLeft: '20vh',
    marginTop: '20vh',
    textAlign: 'center'
  },
  buttonDiv: {
    padding: '20px'
  }
})
