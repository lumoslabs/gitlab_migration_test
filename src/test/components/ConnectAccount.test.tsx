import React from 'react'
import ConnectAccount from '@components/ui/ConnectAccount'
import { render } from '@testing-library/react'

const handleClick = () => { window.location.href = '/' }

it('matches snapshot', () => {
  const { asFragment } = render(
    <ConnectAccount handleCancel={handleClick} handleConnect={handleClick}
   />, {}
  )
  expect(asFragment()).toMatchSnapshot()
})