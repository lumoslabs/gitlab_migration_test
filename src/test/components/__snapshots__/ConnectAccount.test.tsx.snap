// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flexColumnAlignCenter_1ndpwxc container"
  >
    <div
      class="flexRowAllCenter_1cn94ih row"
    >
      <p
        class="title_wdqd43"
      >
        By clicking Connect, Google Assistant will collect your Lumosity audio commands, and their Terms and Privacy Policy will also apply along with Lumosity’s.
        <br />
        <br />
        To remove Google Assistant’s access, please visit Lumosity's Help Center.
      </p>
    </div>
    <div
      class="flexRowAllCenter_1cn94ih row"
    >
      <div
        class="flexColumnAllCenter_1vexgx1-o_O-buttonDiv_1rece6a col"
      >
        <button
          class="button_15gg8qi"
        >
          Cancel
        </button>
      </div>
      <div
        class="flexColumnAllCenter_1vexgx1-o_O-buttonDiv_1rece6a col"
      >
        <button
          class="button_15gg8qi"
        >
          Connect
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
